src/lib.rs

```diff
@@ -40,16 +40,27 @@ pub fn parse(contents: &str) -> Result<EditorConfigINIAST, Error<Rule>> {
 							ast.body.push(Item::Section(current_section.unwrap()));
 						}
 						current_section = Some(Section {
-							name: line.into_inner().next().unwrap().as_str().to_string(),
+							header: SectionHeader {
+								name: line.into_inner().next().unwrap().as_str().to_string(),
```

Comment:
If we go the slicing route, we won't have to blindly `next` each token, but we can pull them by name and slice accordingly. Thing is, I'm not sure how to backreference where the whitespace ends. I'd hate to actually parse it myself, seeing as the parser already handles this.

Reply:
If necessary i absolutely agree with you. If we're able to have a referential-version of the AST, i see no problem storing that information as well.